snapcraft/internal/meta/command.py

```diff
@@ -43,62 +37,174 @@
     """Returns the shebang from file_path."""
-    if not os.path.exists(file_path):
-        raise errors.ShebangNotFoundError()
-
     with open(file_path, "rb") as exefile:
         if exefile.read(2) != b"#!":
-            raise errors.ShebangNotFoundError()
+            return None
         shebang_line = exefile.readline().strip().decode("utf-8")
 
-    # posix is set to False to respect the quoting of variables.
-    shebang_parts = shlex.split(shebang_line, posix=False)
-    # remove the leading /usr/bin/env.
-    if shebang_parts[0] == "/usr/bin/env":
-        shebang_parts = shebang_parts[1:]
-    # or if the shebang startswith /.
-    elif shebang_parts[0].startswith("/"):
-        raise errors.ShebangInRoot()
+    return shebang_line
 
-    return shebang_parts
 
+def _find_executable(*, command: str, prime_dir: str) -> Optional[str]:
+    """Find executable in common binary path locations.
 
-def _find_executable(*, binary: str, prime_dir: str) -> str:
-    found_path: Optional[str] = None
+    :returns: path to executable found, whether it is in searched
+              prime directory paths, or on host using shutil.which().
+    """
     binary_paths = (
-        os.path.join(p, binary) for p in common.get_bin_paths(root=prime_dir)
+        os.path.join(p, command) for p in common.get_bin_paths(root=prime_dir)
     )
     for binary_path in binary_paths:
         if _executable_is_valid(binary_path):
-            found_path = binary_path
-            break
-    else:
-        # Last chance to find in the prime_dir, mostly for backwards compatibility,
-        # to find the executable, historical snaps like those built with the catkin
-        # plugin will have roslaunch in a path like /opt/ros/bin/roslaunch.
-        for root, _, files in os.walk(prime_dir):
-            if _executable_is_valid(os.path.join(root, binary)):
-                found_path = os.path.join(root, binary)
-                break
+            return binary_path
+
+    # Last chance to find in the prime_dir, mostly for backwards compatibility,
+    # to find the executable, historical snaps like those built with the catkin
+    # plugin will have roslaunch in a path like /opt/ros/bin/roslaunch.
+    for root, _, files in os.walk(prime_dir):
+        if _executable_is_valid(os.path.join(root, command)):
+            return os.path.join(root, command)
+
+    # Finally, check if it is part of the system.
+    return shutil.which(command)
+
+
+def _resolve_snap_command_path(
+    *, command: str, prime_dir: str
+) -> Tuple[Optional[str], bool]:
+    """Resolve given command relative to prime directory.
+
+    For backwards compatibility, search for command in typical binary
+    paths.  This would appear to assume that command is the basename,
+    but prior behavior does not check for this, so it could be found
+    relative to the binary paths.
+
+    :returns: tuple of (resolved_command, search_required).
```

Comment:
if you are going to go with this RsT friendly format, I would suggest you also document command and prime_dir

Reply:
I just standardized on the google guide, but if you have a preference, I'll update it.  I just did it to match the module but figured we should standardize across all... and the google style seems nice.

Reply:
We discussed this some time ago, when the team was different and agreed to following something similar to this https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

I recall it having been in our repo docs in the past, I just cannot seem to find it now :shrug: